monai/transforms/spatial/array.py

```diff
@@ -2523,6 +2523,7 @@ def __call__(
         img = convert_to_tensor(img, track_meta=get_track_meta())
         if lazy_:
             if self._do_transform:
+                self.rand_affine_grid(sp_size, randomize=randomize, lazy=True)
```

Comment:
@wyli I just noticed your commits. I'm going to expand the unit tests further for this PR.
I noticed that you've reintroduced randomize=randomize. Strictly speaking, that randomize always gets called twice if it is called, but I'm presuming this is for backwards compatibility for the RNG outputs